src/firewall-cmd.in

```diff
@@ -972,6 +972,9 @@ if (a.direct and not options_direct) or (options_direct and not a.direct):
     cmd.fail(parser.format_usage() +
              "Wrong usage of 'direct' options.")
 
+if a.zone and a.direct:
+    cmd.fail(parser.format_usage() + "--zone= is an invalid option with --direct")
```

Comment:
The equals sign is optional and may confuse users that don't user it. `--zone foobar` and `--zone=foobar` are both accepted. Better to just use "--zone" here.